lib/record_store/provider.rb

```diff
@@ -150,6 +154,11 @@ def retry_on_connection_errors(
         loop do
           begin
             return yield
+          rescue ProviderUnavailableError
+            raise if max_retries <= 0
+            max_retries -= 1
+
+            waiter.wait
```

Comment:
You probably want to change the message line 148 to encompass both connection reset & provider unavailable. Currently, debug might not be easy otherwise!

Reply:
Good catch!